packages/structure/src/language_server/commands.ts

```diff
@@ -0,0 +1,106 @@
+import { ExecuteCommandOptions } from 'vscode-languageserver'
+import { command_builder } from '../interactive_cli/command_builder'
+import { redwood_gen_dry_run as dry_run } from '../interactive_cli/dry_run'
+import { RedwoodCommandString } from '../interactive_cli/RedwoodCommandString'
+import { VSCodeWindowUI } from '../interactive_cli/ui'
+import { RWProject } from '../model'
+import { lazy, memo } from '../x/decorators'
+import { URL_toFile } from '../x/URL'
+import {
+  FileSet_fromTextDocuments,
+  WorkspaceEdit_fromFileSet,
+} from '../x/vscode-languageserver-types'
+import { RWLanguageServer } from './RWLanguageServer'
+
+export const redwoodjs_commands = {
+  'redwoodjs.cli': 'redwoodjs.cli',
+}
+
+export type CommandID = keyof typeof redwoodjs_commands
+
+export class CommandsManager {
+  constructor(public server: RWLanguageServer) {}
+
+  @lazy() get options(): ExecuteCommandOptions {
+    return {
+      commands: Object.keys(redwoodjs_commands),
+      workDoneProgress: true,
+    }
+  }
+
+  @memo() start() {
+    const { connection } = this.server
+    connection.onExecuteCommand(async (params) => {
+      if (params.command === redwoodjs_commands['redwoodjs.cli']) {
+        const [cmd, cwd] = params.arguments ?? []
+        await this.command__cli(cmd, cwd)
+      }
+    })
+  }
+
+  // --- start command implementations
+  private async command__cli(cmdString?: string, cwd?: string) {
+    const {
+      vscodeWindowMethods,
+      host,
+      projectRoot,
+      connection,
+      documents,
+    } = this.server
+    cwd = cwd ?? projectRoot
+    if (!cwd) return // we need a cwd to run the CLI
+    // parse the cmd. this will do some checks and throw
+    let cmd = new RedwoodCommandString(cmdString ?? '...')
+
+    if (
+      cmd.processed.startsWith('dev --open') ||
```

Comment:
@thedavidprice This is what I mentioned on Slack. Instead of adding the missing commands to the redwood-cli I just added a "TODO" of sorts. This way we can just merge this code, then I can update studio, and we don't need to wait any longer for all the other cool stuff that we've been missing out on. (like seeing diagnostics in VSCode)